src/App/Fossa/Config/Report.hs

```diff
@@ -40,20 +43,45 @@ import Options.Applicative (
   option,
   optional,
   progDesc,
+  strOption,
   switch,
  )
 import Path.IO (getCurrentDir)
+import Text.Read (readMaybe)
 
 data ReportType = Attribution deriving (Eq, Ord, Enum, Bounded)
 
 instance Show ReportType where
   show Attribution = "attribution"
 
--- TODO: Add support for text-format reports
 data ReportOutputFormat
   = ReportJson
-  -- ReportPretty
-  deriving (Eq, Ord, Show)
+  | ReportMarkdown
+  | ReportSpdx
+  deriving (Eq, Ord, Enum, Bounded)
+
+instance Read ReportOutputFormat where
+  readsPrec _ s | s == show ReportJson = [(ReportJson, safeStripPrefix (show ReportJson) s)]
+  readsPrec _ s | s == show ReportSpdx = [(ReportSpdx, safeStripPrefix (show ReportSpdx) s)]
+  readsPrec _ s | s == show ReportMarkdown = [(ReportMarkdown, safeStripPrefix (show ReportMarkdown) s)]
+  readsPrec _ _ = []
+
+safeStripPrefix :: String -> String -> String
```

Comment:
I think this is the same as `Data.List.Extra.dropPrefix`.  I also like the `drop` name more because it evokes that behavior.  `safe` is often used for functions that throw exceptions and converts them into functions returning `Maybe` or `Either`.  This feels like the opposite.